mqtt-v5/src/decoder.rs

```diff
@@ -1293,4 +1287,14 @@ mod tests {
         });
         assert_eq!(with_subscription_identifier_expected, decoded);
     }
+    #[test]
+    fn test_decode_variable_int_crash() {
+        let number: u32 = u32::MAX;
+        assert_eq!(
+            decode_variable_int(&mut Cursor::new(&mut BytesMut::from(
+                number.to_be_bytes().as_slice()
+            ))),
+            Err(DecodeError::InvalidRemainingLength)
+        );
```

Comment:
Sorry, I know `EncodeError` and `DecodeError` contain `std::io::Error` which doesn't derive `PartialEq`, which in turn makes it hard to derive for those errors as well.

It may be easier to use the `matches!()` macro.

```suggestion
        let number: u32 = u32::MAX;
        let result = decode_variable_int(&mut Cursor::new(&mut BytesMut::from(
            number.to_be_bytes().as_slice(),
        )));

        assert!(matches!(result, Err(DecodeError::InvalidRemainingLength)));
```

Reply:
I didn't know about the matches macro, thanks!